runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/graph/ProtoOverridesTest.java

```diff
@@ -0,0 +1,253 @@
+                            .setSpec(FunctionSpec.newBuilder().setUrn("beam:repeated"))
+                            .build())
+                    .putTransforms(
+                        "third",
+                        PTransform.newBuilder()
+                            .setSpec(FunctionSpec.newBuilder().setUrn("beam:repeated"))
+                            .build())
+                    .putPcollections(
+                        "intermediatePc",
+                        PCollection.newBuilder().setUniqueName("intermediate").build())
+                    .putCoders(
+                        "coder",
+                        Coder.newBuilder().setSpec(SdkFunctionSpec.getDefaultInstance()).build()))
+            .build();
+
+    ByteString newPayload = ByteString.copyFrom("foo-bar-baz".getBytes());
+    Pipeline updated =
+        ProtoOverrides.updateTransform(
+            "beam:repeated",
+            p,
+            (transformId, existingComponents) -> {
+              String subtransform = String.format("%s_sub", transformId);
+              return MessageWithComponents.newBuilder()
+                  .setPtransform(
+                      PTransform.newBuilder()
+                          .setSpec(
+                              FunctionSpec.newBuilder()
+                                  .setUrn("beam:repeated:replacement")
+                                  .setPayload(newPayload))
+                          .addSubtransforms(subtransform))
+                  .setComponents(
+                      Components.newBuilder()
+                          .putTransforms(
+                              subtransform,
+                              PTransform.newBuilder().setUniqueName(subtransform).build()))
+                  .build();
+            });
+    PTransform updatedSecond = updated.getComponents().getTransformsOrThrow("second");
+    PTransform updatedThird = updated.getComponents().getTransformsOrThrow("third");
+
+    assertThat(updatedSecond, not(equalTo(p.getComponents().getTransformsOrThrow("second"))));
+    assertThat(updatedThird, not(equalTo(p.getComponents().getTransformsOrThrow("third"))));
+    assertThat(updatedSecond.getSubtransformsList(), contains("second_sub"));
+    assertThat(updatedSecond.getSpec().getPayload(), equalTo(newPayload));
+    assertThat(updatedThird.getSubtransformsList(), contains("third_sub"));
+    assertThat(updatedThird.getSpec().getPayload(), equalTo(newPayload));
+
+    assertThat(updated.getComponents().getTransformsMap(), hasKey("second_sub"));
+    assertThat(updated.getComponents().getTransformsMap(), hasKey("third_sub"));
+    assertThat(
+        updated.getComponents().getTransformsOrThrow("second_sub").getUniqueName(),
+        equalTo("second_sub"));
+    assertThat(
+        updated.getComponents().getTransformsOrThrow("third_sub").getUniqueName(),
+        equalTo("third_sub"));
+  }
+
+  @Test
+  public void replaceExistingCompositeFails() {
```

Comment:
This test name is worded like it expects the operation to fail, but based on the implementation it looks like it's expecting it to succeed. Either the name should be changed or the implementation should be fixed.

Reply:
s/fails/succeeds

Done.